tests/api2/test_306_chart_release.py

```diff
@@ -0,0 +1,436 @@
+    results = GET(f'/chart/release/id/{release_id}/')
+    assert results.status_code == 200, results.text
+    assert isinstance(results.json(), dict), results.text
+    assert results.json()['pod_status']['desired'] == 2, results.text
+
+
+@pytest.mark.dependency(name='hostPath_dataset')
+def test_20_create_datasets_for_ipfs_hostPath(request):
+    depends(request, ['pool_04'], scope='session')
+    result = POST('/pool/dataset/', {'name': f'{pool_name}/ipfs-staging'})
+    assert result.status_code == 200, result.text
+    result = POST('/pool/dataset/', {'name': f'{pool_name}/ipfs-data'})
+    assert result.status_code == 200, result.text
+
+
+@pytest.mark.dependency(name='ipfs_schema_values')
+def test_21_change_some_ipfs_schema_values(request):
+    depends(request, ['release_ipfs', 'hostPath_dataset'])
+    global payload
+    payload = {
+        'values': {
+            'updateStrategy': 'RollingUpdate',
+            'service': {
+                'swarmPort': 10401,
+                'apiPort': 10501,
+                'gatewayPort': 10880
+            },
+            'appVolumeMounts': {
+                'staging': {
+                    'datasetName': 'ix-ipfs-staging',
+                    'mountPath': '/export',
+                    'hostPathEnabled': True,
+                    'hostPath': f'/mnt/{pool_name}/ipfs-staging'
+                },
+                'data': {
+                    'datasetName': 'ix-ipfs-data',
+                    'mountPath': '/data/ipfs',
+                    'hostPathEnabled': True,
+                    'hostPath': f'/mnt/{pool_name}/ipfs-staging'
+                }
+            }
+        }
+    }
+    results = PUT(f'/chart/release/id/{release_id}/', payload)
+    assert results.status_code == 200, results.text
+    job_status = wait_on_job(results.json(), 300)
+    assert job_status['state'] == 'SUCCESS', str(job_status['results'])
+
+
+def test_22_verify_ipfs_updateStrategy(request):
+    depends(request, ['ipfs_schema_values'])
+    results = GET(f'/chart/release/id/{release_id}/')
+    assert results.status_code == 200, results.text
+    assert isinstance(results.json(), dict), results.text
+    assert results.json()['config']['updateStrategy'] == payload['values']['updateStrategy'], results.text
+
+
+@pytest.mark.parametrize('key', ['swarmPort', 'apiPort', 'gatewayPort'])
+def test_23_verify_ipfs_service_port(request, key):
+    depends(request, ['ipfs_schema_values'])
+    results = GET(f'/chart/release/id/{release_id}/')
+    assert results.status_code == 200, results.text
+    assert isinstance(results.json(), dict), results.text
+    assert results.json()['config']['service'][key] == payload['values']['service'][key], results.text
+    assert str(payload['values']['service'][key]) in str(results.json()['used_ports']), results.text
+
+
+@pytest.mark.parametrize('key', ['datasetName', 'mountPath', 'hostPathEnabled', 'hostPath'])
+def test_24_verify_ipfs_appVolumeMounts_staging(request, key):
```

Comment:
We should verify that kubernetes is indeed using host paths specified for staging/data, this can be seen by looking up `resources.host_path_volumes` attribute from output of `chart.release.query`, you would need to specify `{"extra": {"retrieve_resources": true}}` though so that you get resources as well.

Reply:
Not sure I got that right but I do not see resources.

```
In [19]: GET('/chart/release/id/ipfs/', {"extra": {"retrieve_resources": True}}).json()
Out[19]: 
{'name': 'ipfs',
 'info': {'first_deployed': '2021-03-23T06:30:12.165321406-07:00',
  'last_deployed': '2021-03-23T06:30:47.965122984-07:00',
  'deleted': '',
  'description': 'Upgrade complete',
  'status': 'deployed',
  'notes': 'IPFS can be accessed from the following URL:\nhttp://$node_ip:10501/webui\n'},
 'config': {'appVolumeMounts': {'data': {'datasetName': 'ix-ipfs-data',
    'hostPath': '/mnt/tank/ipfs-data',
    'hostPathEnabled': True,
    'mountPath': '/data/ipfs'},
   'staging': {'datasetName': 'ix-ipfs-staging',
    'hostPath': '/mnt/tank/ipfs-staging',
    'hostPathEnabled': True,
    'mountPath': '/export'}},
  'environmentVariables': [],
  'image': {'pullPolicy': 'IfNotPresent',
   'repository': 'ipfs/go-ipfs',
   'tag': 'v0.8.0-rc1'},
  'ixCertificateAuthorities': {},
  'ixCertificates': {},
  'ixChartContext': {},
  'ixExternalInterfacesConfiguration': [],
  'ixExternalInterfacesConfigurationNames': [],
  'ixVolumes': [{'hostPath': '/mnt/tank/ix-applications/releases/ipfs/volumes/ix_volumes/ix-ipfs-staging'},
   {'hostPath': '/mnt/tank/ix-applications/releases/ipfs/volumes/ix_volumes/ix-ipfs-data'}],
  'service': {'apiPort': 10501, 'gatewayPort': 10880, 'swarmPort': 10401},
  'updateStrategy': 'RollingUpdate'},
 'version': 3,
 'namespace': 'ix-ipfs',
 'chart_metadata': {'name': 'ipfs',
  'home': 'https://ipfs.io',
  'sources': ['https://github.com/ipfs/go-ipfs',
   'https://hub.docker.com/r/ipfs/go-ipfs'],
  'version': '1.0.1',
  'description': 'Global, Versioned, peer-to-peer filesystem.',
  'keywords': ['storage', 'p2p'],
  'icon': 'https://ipfs.io/ipfs/QmVk7srrwahXLNmcDYvyUEJptyoxpndnRa57YJ11L4jV26/ipfs.go.png',
  'apiVersion': 'v1',
  'appVersion': 'master',
  'dependencies': [{'name': 'common',
    'version': '2101.0.0',
    'repository': 'file://../../../library/common/2101.0.0',
    'enabled': True}],
  'latest_chart_version': '1.0.1'},
 'id': 'ipfs',
 'catalog': 'OFFICIAL',
 'catalog_train': 'charts',
 'path': '/mnt/tank/ix-applications/releases/ipfs',
 'dataset': 'tank/ix-applications/releases/ipfs',
 'status': 'DEPLOYING',
 'used_ports': [{'port': 10401, 'protocol': 'TCP'},
  {'port': 10501, 'protocol': 'TCP'},
  {'port': 10880, 'protocol': 'TCP'}],
 'pod_status': {'desired': 3, 'available': 0},
 'update_available': False,
 'human_version': 'master_1.0.1',
 'human_latest_version': 'master_1.0.1',
 'container_images_update_available': False,
 'portals': {'web_portal': ['http://192.168.2.200:10501/webui']}
```

Reply:
Done https://github.com/truenas/middleware/pull/6609/commits/9899858a246975728b005ae31e2a9b3d15f1adab